lib/inc/internal/detectors/virtualbox_detector.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <internal/sources/dmi_source.hpp>
+
+namespace whereami { namespace detectors {
+
+    /**
+     * VirtualBox detector function
+     * @param dmi_source An instance of a DMI data source
+     * @return Whether this machine is a VirtualBox guest
```

Comment:
I'm not super familiar with practices in cpp - is it customary to include any sort of type information in these doc strings? I guess its pretty obvious from the definition (unlike ruby).